packages/compiler/test/tsx/meta.ts

```diff
@@ -52,6 +52,7 @@ test('extract scripts', async () => {
 				},
 				type: 'module',
 				content: 'console.log({ test: `literal` })',
+				lang: '',
```

Comment:
Go make this super annoying, it serializes nil strings into an empty string instead of undefined/omitting the property (and there's no omitempty, unlike JSON). 

We'd need two separate structs, but then since there's no inheritance, we'd need to duplicate the definition.

In truth, I kinda wanted to use the same format for both so that if there's ever a future where the language of a script tag is decoupled from its hoisting we already have the shape for it and it wouldn't be breaking (ex: `is:inline` + `lang="ts"` would result in type inline, lang ts)

Reply:
Thanks that's good to know. Not a big deal then as long as our public API hides it. Maybe we could further generalize it with a `props: { ... }` property too (if Go allows that).